File: Visual Basic .NET/Project Euler #010.vb
Module Module1

    Sub Main()
    
        ' Solución para el problema #10 de Project Euler
        ' "Halla la suma de todos los números primos inferiores a dos millones."
        ' https://projecteuler.net/problem=10
        ' Código espagueti de 2015; optimizar

        'El programa suma todos los primos menores de dos millones (2000000)

        Dim divisor As Integer = 1
        Dim dividendo As Integer = 1
        Dim cuenta As Integer = 2
        Dim limite As Integer = 2000000
        Dim suma As Long = 0

        Do Until dividendo > limite

            'Como se ha hecho en programas anteriores, el método de búsqueda es sencillo
            'Se buscan primos sabiendo que son números que no pueden ser divididos por ningún número hasta su raíz cuadrada
            'El programa se detiene hasta alcanzar el límite

            If dividendo Mod divisor <> 0 And divisor > Math.Sqrt(dividendo) Then
                suma += dividendo
                dividendo += 2
                divisor = 3
                cuenta += 1

                'Console.Write(dividendo)
                'Console.Write("  ")
                'Console.Write("  ")
                'Console.Write(cuenta)
                'Console.Write("  ")
                'Console.WriteLine(suma)

            ElseIf dividendo Mod divisor = 0 Then
                dividendo += 2
                divisor = 3
            ElseIf divisor > dividendo Then
                divisor = 3
            Else
                divisor += 2
            End If

        Loop

        Console.Write(dividendo)
        Console.Write("  ")
        Console.Write("  ")
        Console.Write(cuenta)
        Console.Write("  ")
        Console.Write(suma)
        Console.WriteLine()

        Console.Write("La respuesta es ")
        Console.Write(suma)
        Console.ReadLine()

        '2 y 5 no están siendo contados

    End Sub

End Module

File: C/Project Euler #010.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

/* Solución para el problema #10 de Project Euler
 * "Encuentra la suma de todos los números primos inferiores a dos millones."
 * https://projecteuler.net/problem=10            */

int esPrimo(long long int n);

int main()
{
    long long int respuesta = 0;
    for (long long int i = 2; i <= 2000000; i++){
        if  (esPrimo(i) == 1) respuesta += i;
    }

    printf("La respuesta es:%lld\n", respuesta);
    return 0;
}

int esPrimo(long long int n){//Función para determinar si un numero es primo, basada en la Criba de Eratóstenes
    if (n == 1) return 1; ////*Cambiar a 0 o mantener a conveniencia del programa
    else if (n < 4) return 1; //Si son menores de 4, son primos
    else if (n % 2 == 0) return 0; //Si son pares, no son primos
    else if (n < 9) return 1; //Si son menores de 9 e impares, son primos
    else if (n % 3 == 0) return 0; //Si son múltiplos de 3, no son primos
    else{
        int r = floor(sqrt(n));
        int f = 5;
        while (f <= r){
            if (n % f == 0) return 0;
            else if (n % (f + 2) == 0) return 0;
            else f += 6;
        }
        return 1;
    }
}
